app/bundles/CoreBundle/Assets/css/app/less/mixins.less

```diff
@@ -260,6 +260,14 @@
   }
 }
 
+.outline() {
+  outline: 2px solid transparent !important;
+  outline-offset: -2px;
+  &:hover { outline: 2px solid transparent; }
+  &:active,
+  &:focus { outline: 2px solid var(--focus) !important; }
+}
```

Comment:
Same issue with these 2 `!important` values as described before: if you're having some issue with a style not coming through, you need to look at the cause and come up with a fix to override. It will either mean combining selectors or fixing something in the styling that is inherited.

Reply:
The main issue here is that Bootstrap comes with many rules for `outline: 0;` in different ways that break the entire functionality
All button-related selectors that have `outline:0;` :

1. `.open > a`
2. `:focus`
3. `.nav > li > a`
4. `.btn`
5. `a:focus, a:active, a:hover`
6. `.btn:active,.btn.active`
7. `.dropdown-toggle:focus`
8. `.dropdown-menu>.active>a,.dropdown-menu>.active>a:hover,.dropdown-menu>.active>a:focus`
9. `.btn-group .dropdown-toggle:active,.btn-group.open .dropdown-toggle`
10. `.navbar-toggle:focus`
11. `.navbar .navbar-nocollapse .navbar-nav>li>a:focus,#app-header.navbar .navbar-nocollapse .navbar-nav>li>a.active`
12. `.btn,.btn:active
`

I'll apply the mixin without !important to all of them

Reply:
Is there a way to just disable the bootstrap made `outline: 0` so you don't have to worry about it in the rest of your styling? Sounds like Bootstrap is just doing something messy and it should be fixed in some kind of reset file.